plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -60,15 +60,24 @@ export class TeamManager {
         this.instanceSiteUrl = serverConfig.SITE_URL || 'unknown'
     }
 
-    public async fetchTeam(teamId: number): Promise<Team | null> {
+    public async fetchTeam(teamId: number, token?: string): Promise<Team | null>
```

Comment:
Suggestion: Make `fetchTeamByToken` a separate method. 

Keeps the interface cleaner and you don't need to deal with 3 cases.